test/extensions/filters/http/cache/simple_http_cache/simple_http_cache_test.cc

```diff
@@ -372,6 +374,28 @@ TEST_F(SimpleHttpCacheTest, UpdateHeadersDisabledForVaryHeaders) {
   EXPECT_TRUE(expectLookupSuccessWithHeaders(lookup(request_path_1).get(), response_headers_1));
 }
 
+TEST_F(SimpleHttpCacheTest, UpdateHeadersSkipSpecificHeaders) {
```

Comment:
Could you add a unit-level test that calls `SimpleHttpCache::updateHeaders` directly and verifies that every field in `headersNotToUpdate` remains unchanged?

Reply:
I've added a unit test that checks for every field except for `vary`. It's infeasible to test `vary` because we are skipping `update_header` operation as long as it has the vary field. We'll be implementing the `vary` logic soon.